### PR TITLE
Packages: Move sync to a classmapped package

### PR DESCRIPTION
**Pull request by tyxla**
This PR moves sync to a classmapped package as a preparatory task, so we can continue working on it and wrap it as an independent package.

#### Changes proposed in this Pull Request:
* Move sync to a classmapped package

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
Part of the Jetpack DNA project - p1HpG7-70O-p2

#### Testing instructions:
* Checkout this branch.
* Run `rm -rf ./vendor` inside Jetpack.
* Run `composer install` inside Jetpack.
* Verify that pass.
* Verify CI passes on this PR and 100% is green.
* Smoke test Jetpack to verify it still works well in both the admin and the frontend.
* Smoke test Jetpack sync (full sync, incremental sync)

#### Proposed changelog entry for your changes:
* Move sync to a classmapped package.

### CODE REVIEW

**Comment by tyxla** [comment]
This is ready for review @Automattic/poseidon @Automattic/jetpack-crew @gravityrail

File: packages/sync/legacy/class.jetpack-sync-listener.php [hunk 1]
```diff
@@ -158,7 +153,7 @@ function bulk_enqueue_full_sync_actions( $action_name, $args_array ) {
 			 *
 			 * @param array The action parameters
 			 */
-			$args = apply_filters( "jetpack_sync_before_enqueue_$action_name", $args );
+			$args        = apply_filters( "jetpack_sync_before_enqueue_$action_name", $args );
```
Comment (aldavigdis) [inline packages/sync/legacy/class.jetpack-sync-listener.php]:
```suggestion
			$args        = apply_filters(
				"jetpack_sync_before_enqueue_$action_name",
				$args
			);
```

File: packages/sync/legacy/class.jetpack-sync-main.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+/**
+ * Jetpack Sync main class.
+ */
+class Jetpack_Sync_Main {
+	/**
+	 * Initialize the main sync actions.
+	 */
+	public static function init() {
+		// Check for WooCommerce support.
+		add_action( 'plugins_loaded', array( 'Jetpack_Sync_Actions', 'initialize_woocommerce' ), 5 );
```
Comment (aldavigdis) [inline packages/sync/legacy/class.jetpack-sync-main.php]:
```suggestion
		add_action(
			'plugins_loaded',
			array( 'Jetpack_Sync_Actions', 'initialize_woocommerce' ),
			5
		);
```

File: packages/sync/legacy/class.jetpack-sync-main.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+		add_action( 'plugins_loaded', array( 'Jetpack_Sync_Actions', 'initialize_wp_super_cache' ), 5 );
+
+		/*
+		 * Init after plugins loaded and before the `init` action. This helps with issues where plugins init
+		 * with a high priority or sites that use alternate cron.
+		 */
+		add_action( 'plugins_loaded', array( 'Jetpack_Sync_Actions', 'init' ), 90 );
+
+		// We need to define this here so that it's hooked before `updating_jetpack_version` is called.
+		add_action( 'updating_jetpack_version', array( 'Jetpack_Sync_Actions', 'cleanup_on_upgrade' ), 10, 2 );
```
Comment (aldavigdis) [inline packages/sync/legacy/class.jetpack-sync-main.php]:
```suggestion
		add_action(
			'updating_jetpack_version',
			array( 'Jetpack_Sync_Actions', 'cleanup_on_upgrade' ),
			10,
			2
		);
```

File: packages/sync/legacy/class.jetpack-sync-main.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+class Jetpack_Sync_Main {
+	/**
+	 * Initialize the main sync actions.
+	 */
+	public static function init() {
+		// Check for WooCommerce support.
+		add_action( 'plugins_loaded', array( 'Jetpack_Sync_Actions', 'initialize_woocommerce' ), 5 );
+
+		// Check for WP Super Cache.
+		add_action( 'plugins_loaded', array( 'Jetpack_Sync_Actions', 'initialize_wp_super_cache' ), 5 );
```
Comment (aldavigdis) [inline packages/sync/legacy/class.jetpack-sync-main.php]:
```suggestion
		add_action(
			'plugins_loaded',
			array( 'Jetpack_Sync_Actions', 'initialize_wp_super_cache' ),
			5
		);

```

File: packages/sync/legacy/class.jetpack-sync-settings.php [hunk 5]
```diff
@@ -60,7 +58,7 @@ static function get_setting( $setting ) {
 				$value = get_site_option( self::SETTINGS_OPTION_PREFIX . $setting );
 			} else {
 				// On single sites just return the default setting
-				$value = Jetpack_Sync_Defaults::get_default_setting( $setting );
+				$value                            = Jetpack_Sync_Defaults::get_default_setting( $setting );
```
Comment (aldavigdis) [inline packages/sync/legacy/class.jetpack-sync-settings.php]: I would just add a newline below here and not bother too much about aligning the equals sign. (I guess that's the same as reverting the change?

```suggestion
				$value = Jetpack_Sync_Defaults::get_default_setting( $setting );

```

**Review by aldavigdis — DISMISSED** [review]
Left some style change suggestions.

File: packages/sync/legacy/class.jetpack-sync-listener.php (hunk 1 above)
Comment (tyxla) [reply packages/sync/legacy/class.jetpack-sync-listener.php]: Thanks, but I disagree and prefer the original one.

File: packages/sync/legacy/class.jetpack-sync-main.php (hunk 2 above)
Comment (tyxla) [reply packages/sync/legacy/class.jetpack-sync-main.php]: Thanks, but I disagree and prefer the original one as it's more readable and WP-like to me.

File: packages/sync/legacy/class.jetpack-sync-main.php (hunk 4 above)
Comment (tyxla) [reply packages/sync/legacy/class.jetpack-sync-main.php]: Thanks, but I disagree and prefer the original one as it's more readable and WP-like to me.

File: packages/sync/legacy/class.jetpack-sync-main.php (hunk 3 above)
Comment (tyxla) [reply packages/sync/legacy/class.jetpack-sync-main.php]: Thanks, but I disagree and prefer the original one as it's more readable and WP-like to me.

File: packages/sync/legacy/class.jetpack-sync-settings.php (hunk 5 above)
Comment (tyxla) [reply packages/sync/legacy/class.jetpack-sync-settings.php]: This was done automatically by PHPCS fixer, so let's keep it 🤷‍♂️

Comment (aldavigdis) [reply packages/sync/legacy/class.jetpack-sync-settings.php]: Automatic fixes my phpcs don't always make sense and we need to look at the code as humans as well. That's what linting is for.

Comment (tyxla) [reply packages/sync/legacy/class.jetpack-sync-settings.php]: Okay, but I suggest let's fix the source of the problem though. If we don't like how phpcs fixer does it, let's make sure it does it properly. This change will come in anyway from anyone who touches that file and has phpcsfixer setup locally.

Also, I agree with the change. See the next line of code - the equal sign aligns with it after the change.

**Comment by aldavigdis** [comment]
Just for the record, I do disagree about how code that exceeds 80 columns is treated here and how `phpcbf`'s auto fix is treated as something that makes sense without exception.

**Review by aldavigdis — APPROVED** [review]
I am not interested in arguing about coding styles. Approved.

**Comment by tyxla** [comment]
> Just for the record, I do disagree about how code that exceeds 80 columns is treated here and how phpcbf's auto fix is treated as something that makes sense without exception.

As I explained in https://github.com/Automattic/jetpack/pull/12572#discussion_r291149079, the change you referred to made sense to me. 

Also, if we're going to fix the 80 columns problem, let's do it for the entire Jetpack codebase. It's more spread out than in just PR, so it would be nice if we don't block the PR for this. After all, I'm only moving lines here and there, and not introducing any long new lines.

**Pull request merged**